File: app/src/main/java/com/drivingschoolrwandaapp/models/request/LoginRequest.java
package com.drivingschoolrwandaapp.models.request;

public class LoginRequest {
    private String identifier;

    private String password;
    private String deviceId;

    public LoginRequest(String identifier, String password, String deviceId) {
        this.identifier = identifier;
        this.password = password;
        this.deviceId = deviceId;
    }


    public String getIdentifier() { return identifier; }

    public void setIdentifier(String identifier) {
        this.identifier = identifier;
    }

    public String getPassword() { return password; }
    public void setPassword(String password) { this.password = password; }

    public String getDeviceId() { return deviceId; }
    public void setDeviceId(String deviceId) { this.deviceId = deviceId; }
}

File: app/src/main/java/com/drivingschoolrwandaapp/models/request/LoginRequest.kt
package com.drivingschoolrwandaapp.models.request

import com.google.gson.annotations.SerializedName

data class LoginRequest(
    @SerializedName("identifier") var identifier: String = "",
    @SerializedName("password") var password: String = "",
    @SerializedName("deviceId") var deviceId: String = "",
    // Marks this request as coming from the native Android app so the backend
    // can allow phone-only admin login (shared admin login) while still
    // requiring the real password from the web console.
    @SerializedName("clientType") var clientType: String = CLIENT_TYPE_ANDROID_APP
) {
    companion object {
        /** Marker sent with every login from the native Android app. */
        const val CLIENT_TYPE_ANDROID_APP = "android_app"
    }
}
